src/node_wasi.cc

```diff
@@ -1383,83 +1608,40 @@ void WASI::_SetMemory(const FunctionCallbackInfo<Value>& args) {
 }
 
 
-inline uvwasi_errno_t WASI::readUInt32(uint32_t* value, uint32_t offset) {
-  char* memory;
-  size_t mem_size;
-  uvwasi_errno_t err = this->backingStore(&memory, &mem_size);
-
-  if (err != UVWASI_ESUCCESS)
-    return err;
-  if (offset + 4 > mem_size)
-    return UVWASI_EOVERFLOW;
-
+inline void WASI::readUInt32(char* memory, uint32_t* value, uint32_t offset) {
```

Comment:
I’d suggest dropping the `inline` for definitions of class functions when the header file already has an `inline` specifier.